ckanext/data_qld/reporting/helpers/export_helpers.py

```diff
@@ -0,0 +1,114 @@
+
+
+def csv_add_org_metrics(org, start_date, end_date, csv_header_row, row_properties, dict_csv_rows, closing_circumstances,
+                        comment_no_reply_max_days, datarequest_open_max_days):
+    """
+    Add reporting metrics for a specific organisation to the CSV data
+    :param org:
+    :param start_date:
+    :param end_date:
+    :param csv_header_row:
+    :param row_properties:
+    :param dict_csv_rows:
+    :param closing_circumstances:
+    :param comment_no_reply_max_days:
+    :param datarequest_open_max_days:
+    :return:
+    """
+    metrics = helpers.gather_metrics(org.get('id', ''), start_date, end_date, comment_no_reply_max_days,
+                                     datarequest_open_max_days)
+
+    csv_header_row.append(org.get('title', ''))
+
+    for key, settings in row_properties.items():
+        if settings['type'] not in ['complex', 'length']:
+            metric_value = metrics.get(settings['property'], '-')
+            dict_csv_rows[key].append(str(int(metric_value)))
+        elif settings['type'] == 'length':
+            metric_value = len(metrics.get(settings['property'], {}))
+            dict_csv_rows[key].append(str(int(metric_value)))
+        elif settings['type'] == 'complex':
+            metric_value = metrics.get(settings['property'], {})[settings['element']]
+            dict_csv_rows[key].append(str(int(metric_value)))
+
+    # Output the closing circumstances rows:
+    datarequest_metrics = metrics.get('datarequests', {})
+
+    circumstance_metrics = datarequest_metrics.get('circumstances', {})
+    no_circumstance_closures = datarequest_metrics.get('no_circumstance', {})
+
+    for circumstance in closing_circumstances:
+        metric_dict = circumstance_metrics.get(circumstance, {})
+        dict_csv_rows['Closed data requests - %s' % circumstance].append(str(int(metric_dict.get('count', 0))))
+
+    for n in no_circumstance_closures:
+        dict_csv_rows['Closed data requests - Closed %s' % n.replace('_', ' ')].append(
+            str(int(no_circumstance_closures[n]['count'])))
+
+    # Now the average closing time for each circumstance
+    for circumstance in closing_circumstances:
+        metric_dict = circumstance_metrics.get(circumstance, {})
+        dict_csv_rows['Average days closed data request - %s' % circumstance].append(str(int(metric_dict.get('average', 0))))
+
+    # Now the average closing time for no circumstance data request closures
+    for n in no_circumstance_closures:
+        dict_csv_rows['Average days closed data request - Closed %s' % n.replace('_', ' ')].append(
+            str(int(no_circumstance_closures[n].get('average', 0))))
+
+    dict_csv_rows['Average days closed data requests - overall'].append(str(int(datarequest_metrics.get('average_overall', 0))))
+
+
+def output_report_csv(csv_header_row, row_order, dict_csv_rows):
+    filename = '%s-report.csv' % datetime.now().isoformat("-")
+
+    with open('/tmp/' + filename, 'wb') as csvfile:
```

Comment:
Temp directory should not be hardcoded, there are APIs for this (eg `tempfile.tempdir`).

Reply:
Changed as requested.